runs/models.tsp

```diff
@@ -483,4 +501,163 @@ model RunToolCallObject {
     /** The arguments that the model expects you to pass to the function. */
     arguments: string;
   }
+}
+
+@oneOf
+@extension("x-oaiExpandable", true)
+union RunStepDeltaStepDetailsObject {
+  RunStepDeltaStepDetailsMessageCreationObject,
+  RunStepDeltaStepDetailsToolCallsObject,
+}
+
+/** Represents a run step delta i.e. any changed fields on a run step during streaming. */
+model RunStepDeltaObject {
+  /** The identifier of the run step, which can be referenced in API endpoints. */
+  id: string;
+
+  /** The object type, which is always `thread.run.step.delta`.
+  object: "thread.run.step.delta";
+
+  /** The delta containing the fields that have changed on the run step. */
+  delta: {
+    /** The details of the run step.  */
+    step_details?: RunStepDeltaStepDetailsObject[];
+  };
+}
+
+/** Represents details of a message creation within a streaming run step. */
+model RunStepDeltaStepDetailsMessageCreationObject {
+  /** The object type, which is always `message_creation`.
+  type: "message_creation";
+
+  /** The message creation data. */
+  message_creation: {
+    /** The ID of the message that was created by this run step. */
+    message_id?: string;
+  };
+}
+
+@oneOf
+@extension("x-oaiExpandable", true)
+union RunStepDeltaStepDetailsToolCallsObject {
+  RunStepDeltaStepDetailsToolCallsCodeObject,
+  RunStepDeltaStepDetailsToolCallsRetrievalObject,
+  RunStepDeltaStepDetailsToolCallsFunctionObject,
+}
+
+/** Represents details of tool calls within a streaming run step. */
+model RunStepDeltaDetailsToolCallsObject {
+  /** The object type, which is always `tool_calls`.
+  type: "tool_calls";
+
+  /**
+   * An array of tool calls the run step was involved in. These can be associated with one of three
+   * types of tools: `code_interpreter`, `retrieval`, or `function`.
+   */
+  tool_calls?: RunStepDeltaStepDetailsToolCallsObject[];
+}
+
+@oneOf
+@extension("x-oaiExpandable", true)
+union RunStepDeltaStepDetailsToolCallsCodeOutput {
+  RunStepDeltaStepDetailsToolCallsCodeOutputLogsObject,
+  RunStepDeltaStepDetailsToolCallsCodeOutputImageObject,
+}
+
+/** Details of a code interpreter tool call that a streaming run step was involved in. */
+model RunStepDeltaStepDetailsToolCallsCodeObject {
+  /** The index of the tool call in the tool calls array. */
+  index: safeint;
+
+  /** The ID of the tool call. */
+  id: string;
+
+  /** The type of the tool call, which is always `code_interpreter`.
+  type: "code_interpreter";
+
+  /** The Code Interpreter tool call definition. */
+  code_interpreter: {
+    /** The input to the Code Interpreter tool call. */
+    input: string;
+
+    /**
+     * The outputs from the Code Interpreter tool call. Code Interpreter can output one or more
+     * items, including text (`logs`) or images (`image`). Each of these are represented by a
+     * different object type.
+     */
+    outputs?: RunStepDeltaStepDetailsToolCallsCodeOutput[];
```

Comment:
The name for this one could be `RunStepDeltaStepDetailsToolCallsCodeObjectCodeInterpreterOutputs`.